src/uController.SourceGenerator/uControllerGenerator.cs

```diff
@@ -335,6 +336,25 @@ IOperation ResolveDeclarationOperation(ISymbol symbol)
                         }
                         populateMetadata.AppendLine($@"                PopulateMetadata<{p.ParameterType}>(parameterInfos[{p.Index}], builder);");
                     }
+
+                    if (p.FromBody)
+                    {
+                        populateMetadata.AppendLine($@"                builder.Metadata.Add(new AcceptsTypeMetadata(typeof({p.ParameterType}), true, new[] {{ ""application/json"" }}));");
+                    }
+                    else if (p.BodyOrService)
+                    {
+                        if (!generatedIsServiceProvider)
+                        {
+                            metadataPreReqs.AppendLine($@"                var ispis = builder.ApplicationServices.GetService<IServiceProviderIsService>();");
+                            generatedIsServiceProvider = true;
+                        }
+                        populateMetadata.AppendLine($@"                   if (!(ispis?.IsService(typeof({p.ParameterType})) ?? false)) builder.Metadata.Add(new AcceptsTypeMetadata(typeof({p.ParameterType}), true, new[] {{ ""application/json"" }}));");
```

Comment:
This is breaking my brain:

```suggestion
                        populateMetadata.AppendLine($@"                   if (ispis?.IsService(typeof({p.ParameterType}) == false) builder.Metadata.Add(new AcceptsTypeMetadata(typeof({p.ParameterType}), true, new[] {{ ""application/json"" }}));");
```